app/src/test/java/com/rickyhu/hushkeyboard/ui/HomeScreenUiTest.kt

```diff
@@ -32,4 +39,46 @@ class HomeScreenUiTest {
             .assertIsEnabled()
             .assertHasClickAction()
     }
+
+    @Test
+    fun `Select input button should exist, should be enable, and should have click action`() {
+        composeTestRule
+            .onNodeWithText("Select input method")
+            .assertExists()
+            .assertIsEnabled()
+            .assertHasClickAction()
+    }
+
+    @Test
+    fun `The textField should be displayed and should be enabled to enter text`() {
+        composeTestRule
+            .onNodeWithText("Type here")
+            .assertIsDisplayed()
+            .performTextInput("Test input")
+
+        composeTestRule.onNodeWithText("Test input").assertIsDisplayed()
+    }
+
+    @Test
+    fun `Settings button should exist, should be enabled, and should have click action`() {
+        composeTestRule
+            .onNodeWithContentDescription("Settings")
+            .assertExists()
+            .assertIsEnabled()
+            .assertIsDisplayed()
+            .assertHasClickAction()
+    }
+
+    @Test
+    fun `Settings button should disappear and action should be invoked upon click`() {
+        composeTestRule
+            .onNodeWithContentDescription("Settings")
+            .performClick()
+
+        composeTestRule
+            .onNodeWithText("Settings")
```

Comment:
This part of the code will be removed because the Settings button won't disappear in this compose test rule. This behavior is normal since the click action will only route the home screen to the SettingScreen,  the button won't disappear while the compose changes.